File: .sling/libs/sushi/eqela/jk-data/0.27.0/jk.data.tests.lua
jk = jk or {}

jk.data = jk.data or {}

jk.data.tests = jk.data.tests or {}

jk.data.tests.TestMain = {}
jk.data.tests.TestMain.__index = jk.data.tests.TestMain
_vm:set_metatable(jk.data.tests.TestMain, {})

function jk.data.tests.TestMain._create()
	local v = _vm:set_metatable({}, jk.data.tests.TestMain)
	return v
end

function jk.data.tests.TestMain:_init()
	self._isClassInstance = true
	self._qualifiedClassName = self._qualifiedClassName or 'jk.data.tests.TestMain'
	self['_isType.jk.data.tests.TestMain'] = true
end

function jk.data.tests.TestMain:_construct0()
	jk.data.tests.TestMain._init(self)
	return self
end

function jk.data.tests.TestMain:_main(args)
	do _g.jk.data.tests.SHAEncoderTests._construct0(_g.jk.data.tests.SHAEncoderTests._create()):execute() end
	do _g.jk.data.tests.MD5EncoderTests._construct0(_g.jk.data.tests.MD5EncoderTests._create()):execute() end
	do _g.jk.data.tests.Base64Tests._construct0(_g.jk.data.tests.Base64Tests._create()):execute() end
	do _g.jk.data.tests.CharacterDecoderTests._construct0(_g.jk.data.tests.CharacterDecoderTests._create()):execute() end
	do return 0 end
end

jk.data.tests.MD5EncoderTests = _g.jk.test.UnitTest._create()
jk.data.tests.MD5EncoderTests.__index = jk.data.tests.MD5EncoderTests
_vm:set_metatable(jk.data.tests.MD5EncoderTests, {
	__index = _g.jk.test.UnitTest
})

function jk.data.tests.MD5EncoderTests._create()
	local v = _vm:set_metatable({}, jk.data.tests.MD5EncoderTests)
	return v
end

function jk.data.tests.MD5EncoderTests:_init()
	self._isClassInstance = true
	self._qualifiedClassName = self._qualifiedClassName or 'jk.data.tests.MD5EncoderTests'
	self['_isType.jk.data.tests.MD5EncoderTests'] = true
end

function jk.data.tests.MD5EncoderTests:_construct0()
	jk.data.tests.MD5EncoderTests._init(self)
	do _g.jk.test.UnitTest._construct0(self) end
	return self
end

function jk.data.tests.MD5EncoderTests:testMD5Encoder()
	if not (_g.jk.md5.MD5Encoder:encodeBuffer(_g.jk.lang.String:toUTF8Buffer("helloWorld")) == "1a833da63a6b7e20098dae06d06602e1") then
		_vm:throw_error(_g.jk.test.FailedAssertionException:instance("UnitTest.sling", "40", "79", nil))
	end
end

function jk.data.tests.MD5EncoderTests:execute()
	do _g.jk.log.Log:debug(self.ctx, "Executing test: " .. "UnitTest" .. " / " .. "MD5Encoder Method") end
	do
		local _status, _err = _vm:execute_protected_call(function()
			do self:testMD5Encoder() end
		end)
		if _err then
			if (_vm:to_table_with_key(_err, '_isType.jk.test.FailedAssertionException') ~= nil) then
				local e = _err
				do _g.jk.log.Log:error(self.ctx, e:toString()) end
			else
				_vm:throw_error(_err)
			end
		end
	end
	do _g.jk.log.Log:debug(self.ctx, "Done executing test: " .. "UnitTest" .. " / " .. "MD5Encoder Method") end
end

jk.data.tests.Base64Tests = _g.jk.test.UnitTest._create()
jk.data.tests.Base64Tests.__index = jk.data.tests.Base64Tests
_vm:set_metatable(jk.data.tests.Base64Tests, {
	__index = _g.jk.test.UnitTest
})

function jk.data.tests.Base64Tests._create()
	local v = _vm:set_metatable({}, jk.data.tests.Base64Tests)
	return v
end

function jk.data.tests.Base64Tests:_init()
	self._isClassInstance = true
	self._qualifiedClassName = self._qualifiedClassName or 'jk.data.tests.Base64Tests'
	self['_isType.jk.data.tests.Base64Tests'] = true
end

function jk.data.tests.Base64Tests:_construct0()
	jk.data.tests.Base64Tests._init(self)
	do _g.jk.test.UnitTest._construct0(self) end
	return self
end

function jk.data.tests.Base64Tests:testBase64Encoding()
	local str = "Hello World"
	if not (_g.jk.base64.Base64Encoder:encodeString(str) == "SGVsbG8gV29ybGQ=") then
		_vm:throw_error(_g.jk.test.FailedAssertionException:instance("UnitTest.sling", "40", "79", nil))
	end
end

function jk.data.tests.Base64Tests:testBase64Decoding()
	local str = "Hello World"
	local encodedStr = _g.jk.base64.Base64Encoder:encodeString(str)
	local decodedStr = _g.jk.base64.Base64Decoder:decode(encodedStr)
	if not (_g.jk.lang.String:forUTF8Buffer(decodedStr) == "Hello World") then
		_vm:throw_error(_g.jk.test.FailedAssertionException:instance("UnitTest.sling", "40", "79", nil))
	end
end

function jk.data.tests.Base64Tests:execute()
	do _g.jk.log.Log:debug(self.ctx, "Executing test: " .. "UnitTest" .. " / " .. "encode string") end
	do
		local _status, _err = _vm:execute_protected_call(function()
			do self:testBase64Encoding() end
		end)
		if _err then
			if (_vm:to_table_with_key(_err, '_isType.jk.test.FailedAssertionException') ~= nil) then
				local e = _err
				do _g.jk.log.Log:error(self.ctx, e:toString()) end
			else
				_vm:throw_error(_err)
			end
		end
	end
	do _g.jk.log.Log:debug(self.ctx, "Done executing test: " .. "UnitTest" .. " / " .. "encode string") end
	do _g.jk.log.Log:debug(self.ctx, "Executing test: " .. "UnitTest" .. " / " .. "decode string") end
	do
		local _status, _err = _vm:execute_protected_call(function()
			do self:testBase64Encoding() end
		end)
		if _err then
			if (_vm:to_table_with_key(_err, '_isType.jk.test.FailedAssertionException') ~= nil) then
				local e = _err
				do _g.jk.log.Log:error(self.ctx, e:toString()) end
			else
				_vm:throw_error(_err)
			end
		end
	end
	do _g.jk.log.Log:debug(self.ctx, "Done executing test: " .. "UnitTest" .. " / " .. "decode string") end
end

jk.data.tests.CharacterDecoderTests = _g.jk.test.UnitTest._create()
jk.data.tests.CharacterDecoderTests.__index = jk.data.tests.CharacterDecoderTests
_vm:set_metatable(jk.data.tests.CharacterDecoderTests, {
	__index = _g.jk.test.UnitTest
})

function jk.data.tests.CharacterDecoderTests._create()
	local v = _vm:set_metatable({}, jk.data.tests.CharacterDecoderTests)
	return v
end

function jk.data.tests.CharacterDecoderTests:_init()
	self._isClassInstance = true
	self._qualifiedClassName = self._qualifiedClassName or 'jk.data.tests.CharacterDecoderTests'
	self['_isType.jk.data.tests.CharacterDecoderTests'] = true
end

function jk.data.tests.CharacterDecoderTests:_construct0()
	jk.data.tests.CharacterDecoderTests._init(self)
	do _g.jk.test.UnitTest._construct0(self) end
	return self
end

function jk.data.tests.CharacterDecoderTests:testCharacterDecoderMethods()
	local dmap = _g.jk.lang.DynamicMap._construct0(_g.jk.lang.DynamicMap._create())
	do dmap:setString("data", "😀 Smile") end
	do
		local str = _g.jk.json.JSONEncoder:encode(dmap, true, false)
		local parse = _vm:to_table_with_key(_g.jk.json.JSONParser:parseBuffer(_g.jk.lang.String:toUTF8Buffer(str)), '_isType.jk.lang.DynamicMap')
		if not (dmap:getString("data", nil) == parse:getString("data", nil)) then
			_vm:throw_error(_g.jk.test.FailedAssertionException:instance("UnitTest.sling", "40", "79", nil))
		end
	end
end

function jk.data.tests.CharacterDecoderTests:execute()
	do _g.jk.log.Log:debug(self.ctx, "Executing test: " .. "UnitTest" .. " / " .. "CharacterDecoder Methods") end
	do
		local _status, _err = _vm:execute_protected_call(function()
			do self:testCharacterDecoderMethods() end
		end)
		if _err then
			if (_vm:to_table_with_key(_err, '_isType.jk.test.FailedAssertionException') ~= nil) then
				local e = _err
				do _g.jk.log.Log:error(self.ctx, e:toString()) end
			else
				_vm:throw_error(_err)
			end
		end
	end
	do _g.jk.log.Log:debug(self.ctx, "Done executing test: " .. "UnitTest" .. " / " .. "CharacterDecoder Methods") end
end

jk.data.tests.SHAEncoderTests = _g.jk.test.UnitTest._create()
jk.data.tests.SHAEncoderTests.__index = jk.data.tests.SHAEncoderTests
_vm:set_metatable(jk.data.tests.SHAEncoderTests, {
	__index = _g.jk.test.UnitTest
})

function jk.data.tests.SHAEncoderTests._create()
	local v = _vm:set_metatable({}, jk.data.tests.SHAEncoderTests)
	return v
end

function jk.data.tests.SHAEncoderTests:_init()
	self._isClassInstance = true
	self._qualifiedClassName = self._qualifiedClassName or 'jk.data.tests.SHAEncoderTests'
	self['_isType.jk.data.tests.SHAEncoderTests'] = true
end

function jk.data.tests.SHAEncoderTests:_construct0()
	jk.data.tests.SHAEncoderTests._init(self)
	do _g.jk.test.UnitTest._construct0(self) end
	return self
end

function jk.data.tests.SHAEncoderTests:testSHAEncoder()
	local shaEncoder = _g.jk.sha.SHAEncoder:create()
	if not (shaEncoder ~= nil) then
		_vm:throw_error(_g.jk.test.FailedAssertionException:instance("UnitTest.sling", "40", "79", nil))
	end
	if not (_g.jk.lang.Buffer:getSize(shaEncoder:encodeAsBuffer(_g.jk.lang.String:toUTF8Buffer("Hello World"), _g.jk.sha.SHAEncoder.SHA1)) == 20) then
		_vm:throw_error(_g.jk.test.FailedAssertionException:instance("UnitTest.sling", "40", "79", nil))
	end
	if not (_g.jk.lang.Buffer:getSize(shaEncoder:encodeAsBuffer(_g.jk.lang.String:toUTF8Buffer("Hello World"), _g.jk.sha.SHAEncoder.SHA256)) == 32) then
		_vm:throw_error(_g.jk.test.FailedAssertionException:instance("UnitTest.sling", "40", "79", nil))
	end
	if not (_g.jk.lang.Buffer:getSize(shaEncoder:encodeAsBuffer(_g.jk.lang.String:toUTF8Buffer("Hello World"), _g.jk.sha.SHAEncoder.SHA384)) == 48) then
		_vm:throw_error(_g.jk.test.FailedAssertionException:instance("UnitTest.sling", "40", "79", nil))
	end
	if not (_g.jk.lang.Buffer:getSize(shaEncoder:encodeAsBuffer(_g.jk.lang.String:toUTF8Buffer("Hello World"), _g.jk.sha.SHAEncoder.SHA512)) == 64) then
		_vm:throw_error(_g.jk.test.FailedAssertionException:instance("UnitTest.sling", "40", "79", nil))
	end
	if not (shaEncoder:encodeAsString(_g.jk.lang.String:toUTF8Buffer("Hello World"), _g.jk.sha.SHAEncoder.SHA1) == "0a4d55a8d778e5022fab701977c5d840bbc486d0") then
		_vm:throw_error(_g.jk.test.FailedAssertionException:instance("UnitTest.sling", "40", "79", nil))
	end
	if not (shaEncoder:encodeAsString(_g.jk.lang.String:toUTF8Buffer("Hello World"), _g.jk.sha.SHAEncoder.SHA256) == "a591a6d40bf420404a011733cfb7b190d62c65bf0bcda32b57b277d9ad9f146e") then
		_vm:throw_error(_g.jk.test.FailedAssertionException:instance("UnitTest.sling", "40", "79", nil))
	end
	if not (shaEncoder:encodeAsString(_g.jk.lang.String:toUTF8Buffer("Hello World"), _g.jk.sha.SHAEncoder.SHA384) == "99514329186b2f6ae4a1329e7ee6c610a729636335174ac6b740f9028396fcc803d0e93863a7c3d90f86beee782f4f3f") then
		_vm:throw_error(_g.jk.test.FailedAssertionException:instance("UnitTest.sling", "40", "79", nil))
	end
	if not (shaEncoder:encodeAsString(_g.jk.lang.String:toUTF8Buffer("Hello World"), _g.jk.sha.SHAEncoder.SHA512) == "2c74fd17edafd80e8447b0d46741ee243b7eb74dd2149a0ab1b9246fb30382f27e853d8585719e0e67cbda0daa8f51671064615d645ae27acb15bfb1447f459b") then
		_vm:throw_error(_g.jk.test.FailedAssertionException:instance("UnitTest.sling", "40", "79", nil))
	end
end

function jk.data.tests.SHAEncoderTests:execute()
	do _g.jk.log.Log:debug(self.ctx, "Executing test: " .. "UnitTest" .. " / " .. "SHAEncoder functions") end
	do
		local _status, _err = _vm:execute_protected_call(function()
			do self:testSHAEncoder() end
		end)
		if _err then
			if (_vm:to_table_with_key(_err, '_isType.jk.test.FailedAssertionException') ~= nil) then
				local e = _err
				do _g.jk.log.Log:error(self.ctx, e:toString()) end
			else
				_vm:throw_error(_err)
			end
		end
	end
	do _g.jk.log.Log:debug(self.ctx, "Done executing test: " .. "UnitTest" .. " / " .. "SHAEncoder functions") end
end

function _main(args)
	do return jk.data.tests.TestMain:_main(args) end
end

File: .sling/libs/web5/eqela/jk-data/0.32.0/jk.data.tests.js
let JkDataTestsSHAEncoderTests = function() {
	JkTestUnitTest.call(this);
};

JkDataTestsSHAEncoderTests.prototype = (function(o) { function F() {}; F.prototype = o; return(new F()); }) (JkTestUnitTest.prototype);
JkDataTestsSHAEncoderTests.prototype.constructor = JkDataTestsSHAEncoderTests;
JkDataTestsSHAEncoderTests.prototype._t = {
	"JkDataTestsSHAEncoderTests" : true,
	"JkTestUnitTest" : true
};
JkDataTestsSHAEncoderTests.prototype._tobj = JkDataTestsSHAEncoderTests;

JkDataTestsSHAEncoderTests.NEW = function() {
	var v = new JkDataTestsSHAEncoderTests;
	return v.CTOR_JkDataTestsSHAEncoderTests();
};

JkDataTestsSHAEncoderTests.prototype.CTOR_JkDataTestsSHAEncoderTests = function() {
	if(JkTestUnitTest.prototype.CTOR_JkTestUnitTest.call(this) == null) {
		return null;
	}
	return this;
};

JkDataTestsSHAEncoderTests.prototype.testSHAEncoder = function() {
	var shaEncoder = JkShaSHAEncoder.create();
	if(!(shaEncoder != null)) {
		throw JkTestFailedAssertionException.instance("SHAEncoderTests.sling", "34", "2", null);
	}
	if(!(JkLangBuffer.getSize((shaEncoder.encodeAsBuffer((JkLangString.toUTF8Buffer("Hello World")), JkShaSHAEncoder.SHA1))) == 20)) {
		throw JkTestFailedAssertionException.instance("SHAEncoderTests.sling", "35", "2", null);
	}
	if(!(JkLangBuffer.getSize((shaEncoder.encodeAsBuffer((JkLangString.toUTF8Buffer("Hello World")), JkShaSHAEncoder.SHA256))) == 32)) {
		throw JkTestFailedAssertionException.instance("SHAEncoderTests.sling", "36", "2", null);
	}
	if(!(JkLangBuffer.getSize((shaEncoder.encodeAsBuffer((JkLangString.toUTF8Buffer("Hello World")), JkShaSHAEncoder.SHA384))) == 48)) {
		throw JkTestFailedAssertionException.instance("SHAEncoderTests.sling", "37", "2", null);
	}
	if(!(JkLangBuffer.getSize((shaEncoder.encodeAsBuffer((JkLangString.toUTF8Buffer("Hello World")), JkShaSHAEncoder.SHA512))) == 64)) {
		throw JkTestFailedAssertionException.instance("SHAEncoderTests.sling", "38", "2", null);
	}
	if(!(shaEncoder.encodeAsString((JkLangString.toUTF8Buffer("Hello World")), JkShaSHAEncoder.SHA1) == "0a4d55a8d778e5022fab701977c5d840bbc486d0")) {
		throw JkTestFailedAssertionException.instance("SHAEncoderTests.sling", "39", "2", null);
	}
	if(!(shaEncoder.encodeAsString((JkLangString.toUTF8Buffer("Hello World")), JkShaSHAEncoder.SHA256) == "a591a6d40bf420404a011733cfb7b190d62c65bf0bcda32b57b277d9ad9f146e")) {
		throw JkTestFailedAssertionException.instance("SHAEncoderTests.sling", "40", "2", null);
	}
	if(!(shaEncoder.encodeAsString((JkLangString.toUTF8Buffer("Hello World")), JkShaSHAEncoder.SHA384) == "99514329186b2f6ae4a1329e7ee6c610a729636335174ac6b740f9028396fcc803d0e93863a7c3d90f86beee782f4f3f")) {
		throw JkTestFailedAssertionException.instance("SHAEncoderTests.sling", "41", "2", null);
	}
	if(!(shaEncoder.encodeAsString((JkLangString.toUTF8Buffer("Hello World")), JkShaSHAEncoder.SHA512) == "2c74fd17edafd80e8447b0d46741ee243b7eb74dd2149a0ab1b9246fb30382f27e853d8585719e0e67cbda0daa8f51671064615d645ae27acb15bfb1447f459b")) {
		throw JkTestFailedAssertionException.instance("SHAEncoderTests.sling", "42", "2", null);
	}
};

JkDataTestsSHAEncoderTests.prototype.execute = function() {
	var v = true;
	JkLogLog.debug(this.ctx, "Executing test: " + ("SHAEncoderTests") + (" / ") + ("SHAEncoder functions"));
	try {
		this.testSHAEncoder();
	}
	catch(e) {
		JkLogLog.error(this.ctx, (e.toString()));
		v = false;
	}
	JkLogLog.debug(this.ctx, "Done executing test: " + ("SHAEncoderTests") + (" / ") + ("SHAEncoder functions"));
	return v;
};

JkDataTestsSHAEncoderTests.IS_INSTANCE = function(o) {
	return o != null && o._t != null && o._t["JkDataTestsSHAEncoderTests"] === true;
};

let JkDataTestsCharacterDecoderTests = function() {
	JkTestUnitTest.call(this);
};

JkDataTestsCharacterDecoderTests.prototype = (function(o) { function F() {}; F.prototype = o; return(new F()); }) (JkTestUnitTest.prototype);
JkDataTestsCharacterDecoderTests.prototype.constructor = JkDataTestsCharacterDecoderTests;
JkDataTestsCharacterDecoderTests.prototype._t = {
	"JkTestUnitTest" : true,
	"JkDataTestsCharacterDecoderTests" : true
};
JkDataTestsCharacterDecoderTests.prototype._tobj = JkDataTestsCharacterDecoderTests;

JkDataTestsCharacterDecoderTests.NEW = function() {
	var v = new JkDataTestsCharacterDecoderTests;
	return v.CTOR_JkDataTestsCharacterDecoderTests();
};

JkDataTestsCharacterDecoderTests.prototype.CTOR_JkDataTestsCharacterDecoderTests = function() {
	if(JkTestUnitTest.prototype.CTOR_JkTestUnitTest.call(this) == null) {
		return null;
	}
	return this;
};

JkDataTestsCharacterDecoderTests.prototype.testCharacterDecoderMethods = function() {
	var dmap = JkLangDynamicMap.NEW();
	dmap.setString("data", "😀 Smile");
	var str = JkJsonJSONEncoder.encode(dmap, true, false);
	var parse = (function(o) {
		if(JkLangDynamicMap.IS_INSTANCE && JkLangDynamicMap.IS_INSTANCE(o)) {
			return o;
		}
		return null;
	}.bind(this))((JkJsonJSONParser.parseBuffer((JkLangString.toUTF8Buffer(str)))));
	if(!(dmap.getString("data", null) == parse.getString("data", null))) {
		throw JkTestFailedAssertionException.instance("CharacterDecoderTests.sling", "37", "2", null);
	}
};

JkDataTestsCharacterDecoderTests.prototype.execute = function() {
	var v = true;
	JkLogLog.debug(this.ctx, "Executing test: " + ("CharacterDecoderTests") + (" / ") + ("CharacterDecoder Methods"));
	try {
		this.testCharacterDecoderMethods();
	}
	catch(e) {
		JkLogLog.error(this.ctx, (e.toString()));
		v = false;
	}
	JkLogLog.debug(this.ctx, "Done executing test: " + ("CharacterDecoderTests") + (" / ") + ("CharacterDecoder Methods"));
	return v;
};

JkDataTestsCharacterDecoderTests.IS_INSTANCE = function(o) {
	return o != null && o._t != null && o._t["JkDataTestsCharacterDecoderTests"] === true;
};

let JkDataTestsBase64Tests = function() {
	JkTestUnitTest.call(this);
};

JkDataTestsBase64Tests.prototype = (function(o) { function F() {}; F.prototype = o; return(new F()); }) (JkTestUnitTest.prototype);
JkDataTestsBase64Tests.prototype.constructor = JkDataTestsBase64Tests;
JkDataTestsBase64Tests.prototype._t = {
	"JkDataTestsBase64Tests" : true,
	"JkTestUnitTest" : true
};
JkDataTestsBase64Tests.prototype._tobj = JkDataTestsBase64Tests;

JkDataTestsBase64Tests.NEW = function() {
	var v = new JkDataTestsBase64Tests;
	return v.CTOR_JkDataTestsBase64Tests();
};

JkDataTestsBase64Tests.prototype.CTOR_JkDataTestsBase64Tests = function() {
	if(JkTestUnitTest.prototype.CTOR_JkTestUnitTest.call(this) == null) {
		return null;
	}
	return this;
};

JkDataTestsBase64Tests.prototype.testBase64Encoding = function() {
	var str = "Hello World";
	if(!(JkBase64Base64Encoder.encodeString(str) == "SGVsbG8gV29ybGQ=")) {
		throw JkTestFailedAssertionException.instance("Base64Tests.sling", "34", "2", null);
	}
};

JkDataTestsBase64Tests.prototype.testBase64Decoding = function() {
	var str = "Hello World";
	var encodedStr = JkBase64Base64Encoder.encodeString(str);
	var decodedStr = JkBase64Base64Decoder.decode(encodedStr);
	if(!(JkLangString.forUTF8Buffer(decodedStr) == "Hello World")) {
		throw JkTestFailedAssertionException.instance("Base64Tests.sling", "42", "2", null);
	}
};

JkDataTestsBase64Tests.prototype.execute = function() {
	var v = true;
	JkLogLog.debug(this.ctx, "Executing test: " + ("Base64Tests") + (" / ") + ("encode string"));
	try {
		this.testBase64Encoding();
	}
	catch(e) {
		JkLogLog.error(this.ctx, (e.toString()));
		v = false;
	}
	JkLogLog.debug(this.ctx, "Done executing test: " + ("Base64Tests") + (" / ") + ("encode string"));
	JkLogLog.debug(this.ctx, "Executing test: " + ("Base64Tests") + (" / ") + ("decode string"));
	try {
		this.testBase64Encoding();
	}
	catch(e1) {
		JkLogLog.error(this.ctx, (e1.toString()));
		v = false;
	}
	JkLogLog.debug(this.ctx, "Done executing test: " + ("Base64Tests") + (" / ") + ("decode string"));
	return v;
};

JkDataTestsBase64Tests.IS_INSTANCE = function(o) {
	return o != null && o._t != null && o._t["JkDataTestsBase64Tests"] === true;
};

let JkDataTestsMD5EncoderTests = function() {
	JkTestUnitTest.call(this);
};

JkDataTestsMD5EncoderTests.prototype = (function(o) { function F() {}; F.prototype = o; return(new F()); }) (JkTestUnitTest.prototype);
JkDataTestsMD5EncoderTests.prototype.constructor = JkDataTestsMD5EncoderTests;
JkDataTestsMD5EncoderTests.prototype._t = {
	"JkDataTestsMD5EncoderTests" : true,
	"JkTestUnitTest" : true
};
JkDataTestsMD5EncoderTests.prototype._tobj = JkDataTestsMD5EncoderTests;

JkDataTestsMD5EncoderTests.NEW = function() {
	var v = new JkDataTestsMD5EncoderTests;
	return v.CTOR_JkDataTestsMD5EncoderTests();
};

JkDataTestsMD5EncoderTests.prototype.CTOR_JkDataTestsMD5EncoderTests = function() {
	if(JkTestUnitTest.prototype.CTOR_JkTestUnitTest.call(this) == null) {
		return null;
	}
	return this;
};

JkDataTestsMD5EncoderTests.prototype.testMD5Encoder = function() {
	if(!(JkMd5MD5Encoder.encodeBuffer((JkLangString.toUTF8Buffer("helloWorld"))) == "1a833da63a6b7e20098dae06d06602e1")) {
		throw JkTestFailedAssertionException.instance("MD5EncoderTests.sling", "33", "2", null);
	}
};

JkDataTestsMD5EncoderTests.prototype.execute = function() {
	var v = true;
	JkLogLog.debug(this.ctx, "Executing test: " + ("MD5EncoderTests") + (" / ") + ("MD5Encoder Method"));
	try {
		this.testMD5Encoder();
	}
	catch(e) {
		JkLogLog.error(this.ctx, (e.toString()));
		v = false;
	}
	JkLogLog.debug(this.ctx, "Done executing test: " + ("MD5EncoderTests") + (" / ") + ("MD5Encoder Method"));
	return v;
};

JkDataTestsMD5EncoderTests.IS_INSTANCE = function(o) {
	return o != null && o._t != null && o._t["JkDataTestsMD5EncoderTests"] === true;
};

let JkDataTestsTestMain = function() {};

JkDataTestsTestMain.prototype._t = { "JkDataTestsTestMain" : true };
JkDataTestsTestMain.prototype._tobj = JkDataTestsTestMain;

JkDataTestsTestMain.NEW = function() {
	var v = new JkDataTestsTestMain;
	return v.CTOR_JkDataTestsTestMain();
};

JkDataTestsTestMain.prototype.CTOR_JkDataTestsTestMain = function() {
	return this;
};

JkDataTestsTestMain._main = function(args) {
	var v = true;
	v = v && JkDataTestsSHAEncoderTests.NEW().execute();
	v = v && JkDataTestsMD5EncoderTests.NEW().execute();
	v = v && JkDataTestsBase64Tests.NEW().execute();
	v = v && JkDataTestsCharacterDecoderTests.NEW().execute();
	return v ? 0 : 1;
};

JkDataTestsTestMain._main();

JkDataTestsTestMain.IS_INSTANCE = function(o) {
	return o != null && o._t != null && o._t["JkDataTestsTestMain"] === true;
};
